core/server.go

```diff
@@ -391,7 +391,11 @@ func (m *RpcServer) ControlEnvironment(cxt context.Context, req *pb.ControlEnvir
 	td := eot.Sub(sot)
 
 	if err != nil {
-		env.Sm.SetState("ERROR")
+		err := env.TryTransition(environment.NewGoErrorTransition(m.state.taskman))
```

Comment:
Since you now use it, perhaps it would be prudent to amend the transition table in `environment.go` so that `GO_ERROR` is allowed from all states.